src/rp2/plugin/report/us/open_positions.py

```diff
@@ -0,0 +1,274 @@
+
+            asset_cost_basis = ZERO
+            for current_transaction in computed_data.in_transaction_set:
+                in_transaction = cast(InTransaction, current_transaction)
+                asset_cost_basis += in_transaction.fiat_in_with_fee * (RP2Decimal("1") - computed_data.get_in_lot_sold_percentage(in_transaction))
+
+            # Skip assets that are zero-value / closed positions.
+            if asset_cost_basis == ZERO:
+                continue
+
+            total_cost_basis += asset_cost_basis
+
+        # Loop thru every asset again, but this time we can do reporting.
+        for asset, computed_data in asset_to_computed_data.items():
+            if not isinstance(asset, str):
+                raise RP2TypeError(f"Parameter 'asset' has non-string value {asset}")
+            ComputedData.type_check("computed_data", computed_data)
+
+            # Net cost basis will be the sum of all usd_in_w_fee for an asset after multiplying off the sold percentage.
+            asset_cost_basis = ZERO
+            for current_transaction in computed_data.in_transaction_set:
+                in_transaction = cast(InTransaction, current_transaction)
+                sold_percent: RP2Decimal = computed_data.get_in_lot_sold_percentage(in_transaction)
+                asset_cost_basis += in_transaction.fiat_in_with_fee * (RP2Decimal("1") - sold_percent)
+
+            # Sum the total amount of crypto asset remaining.
+            total_crypto_balance = ZERO
+            for balance_set in computed_data.balance_set:
+                total_crypto_balance += balance_set.final_balance
+
+            # Skip assets that are zero-value / closed positions.
+            if asset_cost_basis == ZERO:
+                continue
+
+            # For report clarity, change how much precision is displayed in the output based on the unit price. The raw value is
+            # included in the output, so if the user desires they can change the cell format in the resulting file.
+            # The default windowing is set up to hopefully give a good user experience for high value cryptos like BTC
+            # all the way through cryptos with minute unit values like SHIB.
+            unit_cost_basis: RP2Decimal = asset_cost_basis / total_crypto_balance
+            unit_data_style: str = "fiat"
+            if _FIAT_UNIT_DATA_STYLE_4_DECIMAL_MINIMUM <= unit_cost_basis < _FIAT_UNIT_DATA_STYLE_2_DECIMAL_MINIMUM:
+                unit_data_style = "fiat_unit_4"
+            elif unit_cost_basis < _FIAT_UNIT_DATA_STYLE_4_DECIMAL_MINIMUM:
+                unit_data_style = "fiat_unit_7"
+
+            # Only 1 record in the AP sheet per asset.
+            input_sheet.append_rows(1)
+            input_row_index: int = row_indexes["Input"]
+            self._fill_cell(input_sheet, input_row_index, 0, asset)
+            self._fill_cell(input_sheet, input_row_index, 1, "See Input Tab", data_style="fiat_unit_7")
```

Comment:
The Asset and Asset - Exchange tab have VLOOKUPs that present whatever is in the Input tab. I didn't use "Enter {asset} price" due to the risk a user would interpret that as literally entering the price into the Asset or Asset - Exchange tab directly, thus overwriting the VLOOKUP formula and causing them additional work they need not do. 

Initially I figured using a default of 0, null, or "See Input Tab" would be less misleading, but I looked, and I could also set those cells to protected to help prevent the cell from being overwritten.

Reply:
I found a solution for this (let's consider line 4 as an example):
* Input tab: set B4 to `Enter Value...`
* Asset tab: set F4 to `=IF(IFNA(MATCH("Enter Value...", Input.B4), 0)=1, "Fill Input Sheet", Input.B4)`

Reply:
Extending the formula's certainly an option too. Usually on ad hoc sheets I try to keep formulas as uncomplicated as possible, but for something generated by software, that's certainly possible.

We'd still need to use VLOOKUP since an asset with multiple holders (or multiple exchanges having balances on the Asset - Exchange tab) would break any relationship between row numbers on tabs. Essentially something like this:

`=IF( VLOOKUP(A4,$Input.A:B, 2, 0)="Enter value...", "Fill Input Sheet", VLOOKUP(A4,$Input.A:B, 2, 0))`

Reply:
> 

I agree that since the output is generated we can use implementation techniques that otherwise it would be best to avoid: in a manually created spreadsheet I would also use VLOOKUP, but in generated files it's fine to use more complex formulas.

As for the `Asset - Exchange` tab, it's just a many to one relationship between (exchange, holder) -> asset. You can use VLOOKUP if you prefer, but I would suggest using a Python dictionary to keep track of asset -> row_number in the Input tab (so that you know which line an asset is generated at in the Input tab) and then use the dictionary to create the formulas of both the `Asset` and the `Asset - Exchange` tabs, without having to resort to VLOOKUP.